lerobot/scripts/push_dataset_to_hub.py

```diff
@@ -147,39 +128,47 @@ def push_videos_to_hub(repo_id: str, videos_dir: str | Path, revision: str | Non
 
 
 def push_dataset_to_hub(
-    data_dir: Path,
-    dataset_id: str,
-    raw_format: str | None,
-    community_id: str,
-    revision: str,
-    dry_run: bool,
-    save_to_disk: bool,
-    tests_data_dir: Path,
-    save_tests_to_disk: bool,
-    fps: int | None,
-    video: bool,
-    batch_size: int,
-    num_workers: int,
-    debug: bool,
+    raw_dir: Path,
+    raw_format: str,
+    repo_id: str,
+    push_to_hub: bool = True,
+    local_dir: Path | None = None,
+    fps: int | None = None,
+    video: bool = True,
+    batch_size: int = 32,
+    num_workers: int = 8,
+    episodes: list[int] | None = None,
+    force_override: bool = False,
+    cache_dir: Path = Path("/tmp"),
+    tests_data_dir: Path | None = None,
 ):
-    repo_id = f"{community_id}/{dataset_id}"
-
-    raw_dir = data_dir / f"{dataset_id}_raw"
-
-    out_dir = data_dir / repo_id
-    meta_data_dir = out_dir / "meta_data"
-    videos_dir = out_dir / "videos"
-
-    tests_out_dir = tests_data_dir / repo_id
-    tests_meta_data_dir = tests_out_dir / "meta_data"
-    tests_videos_dir = tests_out_dir / "videos"
-
-    if out_dir.exists():
-        shutil.rmtree(out_dir)
-
-    if tests_out_dir.exists() and save_tests_to_disk:
-        shutil.rmtree(tests_out_dir)
+    # Check repo_id is well formated
+    if len(repo_id.split("/")) != 2:
+        raise ValueError(
+            f"`repo_id` is expected to contain a community or user id `/` the name of the dataset (e.g. 'lerobot/pusht'), but contains '{repo_id}'."
+        )
 
+    # Robustify when `local_dir` is str instead of Path
+    if local_dir is not None:
+        local_dir = Path(local_dir)
+
+    # Send warning if local_dir isn't well formated
+    if local_dir is not None:
+        user_id, dataset_id = repo_id.split("/")
+        if local_dir.parts[-2] != user_id or local_dir.parts[-1] != dataset_id:
+            warnings.warn(
+                f"`local_dir` is expected to contain a community or user id `/` the name of the dataset that match the `repo_id` (e.g. 'data/lerobot/pusht'), but is {local_dir}.",
+                stacklevel=1,
+            )
+
+    # Check we don't override an existing `local_dir` by mistake
+    if local_dir is not None and local_dir.exists():
+        if force_override:
+            shutil.rmtree(local_dir)
+        else:
+            raise ValueError(f"`local_dir` already exists ({local_dir}). Use `--force-override 1`.")
+
+    # Download the raw dataset if available
     if not raw_dir.exists():
         download_raw(raw_dir, dataset_id)
```

Comment:
I struggled to understand this. This script is doing too many things at once. Either this behavior should be made more explicit, or have it simply raise an error if `raw_dir` doesn't exists.

Ideally though, I would prefer to have a separate and very simple `download_dataset` or `pull_dataset_from_hub` command and point the user here to the command to run. Something like this:
```python
if not raw_dir.exists():
    print(f"{raw_dir} does not exists. Check your paths or run this command to download the raw dataset:")
    print(f"python lerobot/scripts/download_dataset.py {repo_id}_raw {raw_dir}")
    raise NotADirectoryError
```

Reply:
Redesigned download_raw

Added
```python
if not raw_dir.exists():
    raise NotADirectoryError(
        f"{raw_dir} does not exists. Check your paths or run this command to download an existing raw dataset on the hub:"
        f"python lerobot/common/datasets/push_dataset_to_hub/_download_raw.py --raw-dir your/raw/dir --repo-id your/repo/id_raw"
    )
```

Removed mocking in tests